docs/extensions/guides/renderer-extension.md

```diff
@@ -379,117 +379,286 @@ The `uninstall()` method is implemented in the example above by utilizing the [`
 
-
-*********************************************************************
-WIP below!
-*********************************************************************
-
-
-
-
 ### `appPreferences`
 
-The Preferences page is essentially a global page. Extensions can add custom preferences to the Preferences page, thus providing a single location for users to configure global, for Lens and extensions alike.
+The Preferences page is a built-in global page. Extensions can add custom preferences to the Preferences page, thus providing a single location for users to configure global options, for Lens and extensions alike. The following example demonstrates adding a custom preference:
 
 ``` typescript
-import React from "react"
-import { LensRendererExtension } from "@k8slens/extensions"
-import { myCustomPreferencesStore } from "./src/my-custom-preferences-store"
-import { MyCustomPreferenceHint, MyCustomPreferenceInput } from "./src/my-custom-preference"
-
+import { LensRendererExtension } from "@k8slens/extensions";
+import { ExamplePreference, ExamplePreferenceHint, ExamplePreferenceInput } from "./src/example-preference";
+import { observable } from "mobx";
+import React from "react";
 
 export default class ExampleRendererExtension extends LensRendererExtension {
+
+  @observable preference: ExamplePreference = { enabled: false };
+
   appPreferences = [
     {
-      title: "My Custom Preference",
+      title: "Example Preferences",
       components: {
-        Hint: () => <MyCustomPreferenceHint/>,
-        Input: () => <MyCustomPreferenceInput store={myCustomPreferencesStore}/>
+        Input: () => <ExamplePreferenceInput preference={this.preference}/>,
+        Hint: () => <ExamplePreferenceHint/>
       }
     }
   ];
 }
 ```
 
+App preferences are objects matching the `AppPreferenceRegistration` interface. The `title` field specifies the text to show as the heading on the Preferences page. The `components` field specifies two `React.Component` objects defining the interface for the preference. `Input` should specify an interactive input element for your preference and `Hint` should provide descriptive information for the preference, which is shown below the `Input` element. `ExamplePreferenceInput` expects its React props set to an `ExamplePreference` instance, which is how `ExampleRendererExtension` handles the state of the preference input. `ExampleRendererExtension` has the field `preference`, which is provided to `ExamplePreferenceInput` when it is created. In this example `ExamplePreferenceInput`, `ExamplePreferenceHint`, and `ExamplePreference` are defined in `./src/example-preference.tsx`:
+
+``` typescript
```

Comment:
```suggestion
```tsx
```

The above works and helps, at least, when working on the data.

Reply:
I tried this but started backing it out. tsx works in vscode but not in the mkdocs result (it's all monochrome). Unless there's a fix for the mkdocs output?

Reply:
Let me look into that for you then.

Reply:
Okay, after doing some research it seems like vscode should (or maybe will shortly) treat `typescript` as the correct lexar for everything. I think that we should then remove `tsx` from all these snippets.